msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -92,6 +94,24 @@ public String getId() {
             id = mHomeOid;
         }
 
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
```

Comment:
can you elaborate more on why this happens?  Is the issue consistently reproductible?

Reply:
ah okay... this is a mitigation for this broker bug.

`Secondly, the ID token that we have for WPJ accounts may not contain oid claim, therefore, I've updated the PRT request to ask for PROFILE scope as well to ensure we get oid claim in ID Token. (this issue is in both dev and PROD`

Reply:
Could you please add a comment describing the issue above this line? Just so that future devs have a context.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1560/commits/254d3c2ce980c03b384fbf51d4009e99974d2bae